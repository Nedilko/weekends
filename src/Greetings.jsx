function Greetings() {
  return (
    <>
      <h1 className="font-roboto font-light text-center text-5xl text-stone-200 my-10">
        weekends countdown
      </h1>
    </>
  )
}

export default Greetings
