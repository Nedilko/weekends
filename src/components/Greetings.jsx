function Greetings() {
  return (
    <div className="font-thin uppercase text-center text-slate-600 text-7xl mb-10 mt-20">
      Have a beer
    </div>
  )
}

export default Greetings
