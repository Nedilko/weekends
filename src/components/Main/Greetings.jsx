import PropTypes from 'prop-types'

Greetings.propTypes = {
  text: PropTypes.string.isRequired,
}

function Greetings({ text }) {
  return (
    <div className="mt-32 text-center text-7xl font-thin uppercase">{text}</div>
  )
}

export default Greetings
