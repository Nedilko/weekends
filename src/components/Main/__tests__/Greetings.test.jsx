import Greetings from '@components/Main/Greetings'
import SettingsContext from '@store/Settings'
import { render, screen } from '@utils/test-utils'

describe('Greetings', () => {
  it('should have greetings text', () => {
    render(
      <SettingsContext.Provider
        value={{ data: { greetingsText: 'Sample greetings text' } }}
      >
        <Greetings />
      </SettingsContext.Provider>
    )
    expect(screen.getByText(/sample/i)).toBeInTheDocument()
  })

  it('should render greetings component', () => {
    render(
      <SettingsContext.Provider
        value={{ data: { greetingsText: 'Sample greetings text' } }}
      >
        <Greetings />
      </SettingsContext.Provider>
    )
    expect(screen.getByText(/sample/i)).toMatchSnapshot()
  })
})
