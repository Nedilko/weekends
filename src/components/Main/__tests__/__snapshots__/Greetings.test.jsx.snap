// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greetings should render greetings component 1`] = `
<div
  class="mt-32 text-center text-7xl font-thin uppercase"
>
  Sample greetings text
</div>
`;
